### PR TITLE
Add Logstash to the Marketplace UI and ARM template

### LINKED ISSUE #157: Deploy Logstash as part of the template

### PR DESCRIPTION
**Pull request by russcam**
This PR adds Logstash to the Marketplace UI and ARM template. This is an initial implementation, with possibility to expand in future based on feedback and usage.

- Provide the option to deploy a single instance of Logstash, with configurable VM SKU. The default will remain the same and not deploy Logstash.
- Allow configurable JVM heap size.
- Allow a Base-64 encoded string form of a Logstash conf file be provided, to start Logstash with. Expose variables that can be referenced within the conf file

  - `${ELASTICSEARCH_URL}`: the Elasticsearch endpoint
  - `${LOGSTASH_SYSTEM_PASSWORD}`: password of the built-in `logstash_system` user
  - `${ELASTICSEARCH_CACERT}`: the path to the CA cert used to secure the Elasticsearch 

  The value is Base-64 encoded to match how the value is encoded when received from an uploaded file within the UI.

- Allow Logstash keystore password to be specified, for use in Logstash versions that have keystore functionality.
- Allow additional Logstash plugins to be installed.
- Allow additional YAML configuration to append to logstash.yml configured by the template.
- For Logstash 6.3.0+  and for Logstash < 6.3.0 where X-Pack plugin is installed, configure X-Pack monitoring. Document whether monitoring is enabled based on if and how TLS is configured for the Elasticsearch HTTP layer.

Closes #157

### CODE REVIEW

**Comment by russcam** [comment]
Before this PR is merged, a `6.3` branch must be created and the elastic.co docs set to pick up documentation from that branch. This then makes the way to create a `6.4` branch which will include documentation about Logstash (in addition to subsequent PRs that will add support for 6.4.0 and 6.4.1).

**Comment by russcam** [comment]
@jsvd I've requested a review 😃

File: README.md [hunk 1]
```diff
@@ -362,6 +364,39 @@ in conjunction with other parameters.
...
+    <td>Either <code>Yes</code> or <code>No</code> to provision a machine with Logstash installed. If you have opted to also install the X-Pack plugins using <code>xpackPlugins</code>,
+    a trial license for the <a href="https://www.elastic.co/products/stack">commercial Logstash features</a> will be applied and activated.
+    </td><td><code>No</code></td></tr>
+
+  <tr><td>vmSizeLogstash</td><td>string</td>
+    <td>Azure VM size of the Logstash instance. See <a href="https://github.com/elastic/azure-marketplace/blob/master/build/allowedValues.json">this list for supported sizes</a>.
+    <strong>Check that the size you select is <a href="https://azure.microsoft.com/en-au/regions/services/">available in the region you choose</a></strong>.
+    </td><td><code>Standard_D1</code></td></tr>
+
+  <tr><td>logstashHeapSize</td><td>integer</td>
```
Comment (Mpdreamz) [inline README.md]: Do we need to expose this? Since logstash always deploys on a dedicated VM can we just do the right thing™?

Comment (russcam) [reply README.md]: Do you mean the heap size? If so, I wasn't sure what might be a sensible default, beyond the default in JVM.options (which I think is 1GB?). Thoughts?

Comment (Mpdreamz) [reply README.md]: @russcam mind opening an issue for whether we can rid of the parameter?

**Pull request merged**